packages/tools/perftest/src/timer.ts

```diff
@@ -0,0 +1,113 @@
+import { EdDSATicketPCDPackage } from "@pcd/eddsa-ticket-pcd";
+import { ZKEdDSAEventTicketPCDPackage } from "@pcd/zk-eddsa-event-ticket-pcd";
+import { program } from "commander";
+import logSymbols from "log-symbols";
+import {
+  EdDSATicketProveCase,
+  EdDSATicketVerifyCase
+} from "./cases/EdDSATicketTimer";
+import {
+  ZKEdDSAEventTicketProveCase,
+  ZKEdDSAEventTicketVerifyCase
+} from "./cases/ZKEdDSAEventTicketTimer";
+import { DemoCase, TimerCase } from "./types";
+
+const TIME_TEST_CONFIGS: Record<string, Record<string, () => TimerCase>> = {
+  demo: {
+    demo: () => new DemoCase()
+  },
+  [EdDSATicketPCDPackage.name]: {
+    prove: () => new EdDSATicketProveCase(),
+    verify: () => new EdDSATicketVerifyCase()
+  },
+  [ZKEdDSAEventTicketPCDPackage.name]: {
+    prove: () => new ZKEdDSAEventTicketProveCase(),
+    verify: () => new ZKEdDSAEventTicketVerifyCase()
+  }
+};
+
+async function timeOp(
+  opName: string,
+  iterationCount: number,
+  op: () => Promise<void>
+): Promise<void> {
+  const startMS = performance.now();
+  for (let i = 0; i < iterationCount; i++) {
+    await op();
+  }
+  const endMS = performance.now();
+  console.info(
+    `${logSymbols.success}`,
+    `${opName}${iterationCount > 1 ? ` (average of ${iterationCount})` : ""}: ${
+      (endMS - startMS) / iterationCount
+    }ms`
+  );
+}
+
+async function timerDriver(
+  caseMaker: () => TimerCase,
+  iterationCount: number
+): Promise<void> {
+  const timerCase = caseMaker();
+  console.info(`Running timing test of ${timerCase.name}...`);
+  await timeOp("init", 1, () => timerCase.init());
+  await timeOp("setup", 1, () => timerCase.setup());
+  await timeOp("first op", 1, () => timerCase.op());
+  await timeOp("second op", 1, () => timerCase.op());
+  await timeOp("repeat op", iterationCount, () => timerCase.op());
+}
+
+program
+  .command("timer")
+  .description("Time a set of operations for a PCD packge")
+  .argument("<pcd-package>", "Supported PCD package.")
```

Comment:
it may be worth printing the names of the supported pcd packages and their corresponding timed operations as part of this script, maybe if a `--help` argument is provided